workers/unity/Assets/Playground/Scripts/EntityInitialization/EntityGameObjectCreator.cs

```diff
@@ -1,46 +1,66 @@
-using System;
 using System.Collections.Generic;
-using Improbable.Worker;
-using Unity.Entities;
+using System.IO;
+using Generated.Improbable;
+using Generated.Improbable.Transform;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
 using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
 
 namespace Playground
 {
-    public class EntityGameObjectCreator
+    public class EntityGameObjectCreator : IEntityGameObjectCreator
     {
-        private readonly World world;
-        private readonly Dictionary<string, GameObject> cachedPrefabs;
+        private readonly Dictionary<string, Dictionary<string, GameObject>> cachedPrefabsPerWorkerType
+            = new Dictionary<string, Dictionary<string, GameObject>>();
 
-        public EntityGameObjectCreator(World world, Dictionary<string, GameObject> initialCachedPrefabs = null)
+        public GameObject CreateGameObjectForEntity(SpatialOSEntity entity, WorkerSystem worker)
         {
-            this.world = world;
-            cachedPrefabs = initialCachedPrefabs ?? new Dictionary<string, GameObject>();
-        }
+            if (!cachedPrefabsPerWorkerType.TryGetValue(worker.WorkerType, out var cachedPrefabs))
+            {
+                cachedPrefabsPerWorkerType[worker.WorkerType] = cachedPrefabs = new Dictionary<string, GameObject>();
+            }
 
-        public GameObject CreateEntityGameObject(Entity entity, string prefabPath, Vector3 position,
-            Quaternion rotation, EntityId spatialEntityId)
-        {
-            if (!cachedPrefabs.TryGetValue(prefabPath, out var prefab))
+            if (!entity.HasComponent<Metadata.Component>())
+            {
+                return null;
+            }
+
+            var prefabName = entity.GetComponent<Metadata.Component>().EntityType;
+            if (!entity.HasComponent<Transform.Component>())
             {
-                prefab = Resources.Load<GameObject>(prefabPath);
+                return cachedPrefabs[prefabName] = null;
+            }
+
+            var transform = entity.GetComponent<Transform.Component>();
+            var position = new Vector3(transform.Location.X, transform.Location.Y, transform.Location.Z) +
+                worker.Origin;
+            var rotation = new UnityEngine.Quaternion(transform.Rotation.X, transform.Rotation.Y,
+                transform.Rotation.Z, transform.Rotation.W);
+            var workerSpecificPath = Path.Combine("Prefabs", worker.WorkerType, prefabName);
```

Comment:
We discussed elsewhere that we didn't want to enforce playground maintaining two of everything.

Reply:
We are not enforcing maintaining two, if there is a prefab in `Prefabs\Common\` and not elsewhere then that is applied to all workers. I thought we said having the option of different prefabs is not a priority but having a basic solution like this is OK? Since we don't have the SDK annotations to specify a MonoBehaviour should only run on a specific worker, without different prefabs we'd have no way to do anything like that.

Reply:
Fair enough. Criticism retracted. Will resolve my comment.